### DISCUSSION TITLE
Need help with creating collisions dynamically

### DISCUSSION THREAD

**Discussion by mohanvaddi** [Annoraaq/grid-engine#471] [Q&A]
Hi @Annoraaq 

I'm using `grid-engine` plugin in my phaser3 game. but I need help with collisions in dynamic generated maps.

My game has two scenes, TOWN scene and DUNGEON scene. 
- TOWN scene is basically a map that is exported from tiled and being used just like it is shown in one of your example. It has a `collision` layer. and It is a `ge_charLayer`. Collisions are working fine in this scene.

https://github.com/Annoraaq/grid-engine/assets/58596948/573c727c-bea0-44fd-8832-fe5c35e8225c

- DUNGEON scene
 I am procedurally generating a dungeon based on this [medium article](https://itnext.io/modular-game-worlds-in-phaser-3-tilemaps-3-procedural-dungeon-3bc19b841cd) and I'm trying to add `collision` layer and tileset properties like this to the map. and I'm adding the `COLLISION_TILE` where ever needed dynamically.  - (YOU CAN SEE THIS IN THE VIDEO)

<img width="1079" alt="image" src="https://github.com/Annoraaq/grid-engine/assets/58596948/cca54be2-604f-440b-8515-228c98a2c999">

then I'm creating `gridEngine`, tho the grid-engine movement is working in the scene the collisions are not working in this scene. 

Can grid-engine collisions work in my case, I don't see any example for dynamic creation of collisions in grid-engine docs site ? or Is there something I'm missing here ? please let me know.

**Comment by Annoraaq** [comment]
How exactly are you setting the collision properties?
Is the character actually on the "collision" char layer?
In general it should work with procedurally generated maps the same way as with those created in Tiled.


Btw, you have to make sure that the tileCollisionCache is not enabled (it is not by default): https://annoraaq.github.io/grid-engine/api/interfaces/GridEngineConfigHeadless.html#cacheTileCollisions

**Reply by Annoraaq** [reply]
It just came to my mind that when you are adding whole layers after initializing/creating Grid Engine, they might not be considered. So make sure to not add any additional layers after calling 'create'.

**Reply by mohanvaddi** [reply]
I'm adding "collision" character layer as a `layer property` in tiled now. and I'm embedding a collision tileset with tileset properites. 

and I'm not adding any additional layers after calling 'create' and not using `cahceTileCollisions`


I've create a [codesandbox](https://codesandbox.io/p/devbox/holy-leaf-dlnznw?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cltvsqk4s00073b6jpg1ibwv0%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cltvsqk4r00023b6jbyibi2c1%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cltvsqk4r00043b6javtzkqvv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cltvsqk4r00063b6j11ivebmi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B58.584403722463165%252C41.415596277536835%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cltvsqk4r00023b6jbyibi2c1%2522%253A%257B%2522id%2522%253A%2522cltvsqk4r00023b6jbyibi2c1%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522cltvsqk4r00063b6j11ivebmi%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltvsqk4r00053b6jnp5xtkpv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cltvsqk4r00063b6j11ivebmi%2522%252C%2522activeTabId%2522%253A%2522cltvsqk4r00053b6jnp5xtkpv%2522%257D%252C%2522cltvsqk4r00043b6javtzkqvv%2522%253A%257B%2522id%2522%253A%2522cltvsqk4r00043b6javtzkqvv%2522%252C%2522activeTabId%2522%253A%2522cltvsqk4r00033b6j8289rtgv%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltvsqk4r00033b6j8289rtgv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%252C%2522id%2522%253A%2522cltvt0pix00j23b6j9cluttug%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) recreating the issue. please take a look at this.

please check the `src/game/scenes/DungeonScene.ts` 
from line 55-89 and 215-251.

**Reply by Annoraaq** [reply]
One thing that makes me suspicious is that when I call [getTileAt](https://newdocs.phaser.io/docs/3.55.2/Phaser.Tilemaps.TilemapLayer#getTileAt) on that layer, the `properties` prop of that tile is empty. It should, however contain the collision properties from the tileset. Is it possible that `weightedRandomize` does not set the tile properties inherited by the tileset?
